packages/core/src/lib/toast/context.ts

```diff
@@ -0,0 +1,141 @@
+import { setContext, getContext } from 'svelte';
+import {
+  derived,
+  get,
+  writable,
+  type Readable,
+  type Writable,
+} from 'svelte/store';
+
+import { uniqueId } from '$lib/unique-id';
+import { ToastVariant, type ToastVariantType } from '$lib/toast';
+
+import { pausableProgress } from '../notification/pausable-progress';
+
+/** Internal toast context. */
+export interface ToastContext {
+  state: ToastState;
+  toast: Toast;
+}
+
+/** Notification sender. */
+export interface Toast {
+  /** Push a neutral toast with a message. */
+  neutral: (message: string, closeable?: boolean) => void;
+  /** Push an upload toast with a message. */
+  upload: (message: string, closeable?: boolean) => void;
```

Comment:
I feel like `upload` is a pretty specific use case for this, should it be something like `info` or `progress`? As a user, I have no idea what an `upload` toast is supposed to look like or signify.

Reply:
I pulled down the PR and ran the storybook and it looks like this is intended to be the "process" variant we have in the [figma](https://www.figma.com/file/ERHSYDGTXiEQOVGY8ORg6I/PRIME?type=design&node-id=2650-4838&mode=design&t=hht9SlFQ2x7ltRIa-4).

If this is supposed to be the "process" variant, a few notes:
- Progress bar should go left-to-right
- Process variant should end looking like "success"
- Progress bar should have controlled progress such that it reflects the progress of the long-running action (such as uploading a file)

If this is supposed to be the "info" variant:
- Icon should be the info icon
- There should not be a progress bar

Which [variants](https://www.figma.com/file/ERHSYDGTXiEQOVGY8ORg6I/PRIME?type=design&node-id=2650-4838&mode=design&t=hht9SlFQ2x7ltRIa-4) do we need for R2D2?

Reply:
`process` definitely makes sense as a variant name. I am wondering if we even need the process variant right now? The ticket this is attached to is specifically for the success toast, so seems like we could hold off on the `process` variant right now?

Reply:
Will focus this PR on the success variant.